jax/ops/image/extract.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as onp
+
+from jax import lax
+from jax.ops.image import util
+from jax.numpy import lax_numpy as np
+
+
+def extract_image_patches(image, ksizes, strides, rates, padding):
+    # replicating tensorflow's implementation
+    num_dims = image.ndim
+    num_spatial_dims = num_dims - 2
+    data_format = "NHWC"
```

Comment:
You should probably let the user specify the format.

Reply:
The original implementation had it hard coded but I don't see any reason not to allow this to be specified. I've updated the implementation.